DataFormats/Portable/interface/PortableHostCollection.h

```diff
@@ -53,21 +53,57 @@ class PortableHostCollection {
   // default destructor
   ~PortableHostCollection() = default;
 
+  // has a valid buffer?
+  bool isValid() const { return buffer_.has_value(); }
+
+  // a way to access the number of elements without the buffer validity assertion
+  auto size() const { return view_.metadata().size(); }
+
   // access the View
-  View& view() { return view_; }
-  ConstView const& view() const { return view_; }
-  ConstView const& const_view() const { return view_; }
+  View& view() {
+    assert(isValid());
+    return view_;
+  }
```

Comment:
Alternatively the View accessors could be left unchecked, and require the users to explicitly check `isValid()` or/and the column/scalar accessors of the View to be non-`nullptr` when it is not clearly guaranteed that the `PortableCollection` is non-default constructed.

(we could also throw an exception instead of `assert()`, but maybe the use of default-constructed `PortableCollection` could be more of a logic error rather than something that would depend e.g. on the data?)